java/read-symbols-mobile-style-file/src/main/java/com/esri/arcgisruntime/sample/readsymbolsmobilestylefile/MainActivity.java

```diff
@@ -0,0 +1,431 @@
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+  private static final int PERM_REQUEST_CODE = 1;
+  private static final String[] PERMISSIONS = { Manifest.permission.READ_EXTERNAL_STORAGE };
+
+  private MapView mMapView;
+  private GraphicsOverlay mGraphicsOverlay;
+  private RecyclerView mEyesRecyclerView;
+  private RecyclerView mMouthRecyclerView;
+  private RecyclerView mHatRecyclerView;
+  private ImageView mPreviewView;
+  private SymbolAdapter mEyesAdapter;
+  private SymbolAdapter mMouthAdapter;
+  private SymbolAdapter mHatAdapter;
+
+  private SymbolStyleSearchResultObservable symbolStyleSearchResultObservable = new SymbolStyleSearchResultObservable();
+  private String mFaceSymbolKey;
+
+  private SymbolStyle mSymbolStyle;
+
+  private HashMap<String, SymbolStyleSearchResult> mSelectedSymbols = new HashMap<>();
+
+  private Symbol mCurrentMultilayerSymbol;
+
+  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    mMapView = findViewById(R.id.mapView);
+    mEyesRecyclerView = findViewById(R.id.eyesRecyclerView);
+    mMouthRecyclerView = findViewById(R.id.mouthRecyclerView);
+    mHatRecyclerView = findViewById(R.id.hatRecyclerView);
+    mPreviewView = findViewById(R.id.previewView);
+
+    // create a map
+    ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+    // add the map to the map view
+    mMapView.setMap(map);
+
+    // create a graphics overlay to add graphics to and add it to the map view
+    mGraphicsOverlay = new GraphicsOverlay();
+    mMapView.getGraphicsOverlays().add(mGraphicsOverlay);
+
+    // add listener to handle motion events when the user taps on the map view
+    mMapView.setOnTouchListener(
+        new DefaultMapViewOnTouchListener(MainActivity.this, mMapView) {
+          @Override
+          public boolean onSingleTapConfirmed(MotionEvent motionEvent) {
+            addGraphic(mGraphicsOverlay, mapPointFrom(mMapView, motionEvent), mCurrentMultilayerSymbol);
+            return true;
+          }
+        });
+
+    setupRecyclerViews();
+
+    // add observer to Observable to be notified when symbols are loaded
+    symbolStyleSearchResultObservable.addObserver((o, arg) -> {
+      for (SymbolStyleSearchResult symbol : ((SymbolStyleSearchResultObservable) o).getSymbols()) {
+        // these categories are specific to this SymbolStyle
+        switch (symbol.getCategory().toLowerCase(Locale.ROOT)) {
+          case "eyes":
+            mEyesAdapter.addSymbol(symbol);
+            break;
+          case "mouth":
+            mMouthAdapter.addSymbol(symbol);
+            break;
+          case "hat":
+            mHatAdapter.addSymbol(symbol);
+            break;
+          case "face":
+            mFaceSymbolKey = symbol.getKey();
+            break;
+        }
+        Log.d(TAG, symbol.getCategory());
```

Comment:
This is probably quite useful for now, but lets make sure we remove it if we don't need it for the final sample